server/chat-commands/info.ts

```diff
@@ -2622,12 +2628,17 @@ export const commands: ChatCommands = {
 		}
 
 		const [link, comment] = Utils.splitFirst(target, ',');
-
 		let buf;
 		const YouTube = new YoutubeInterface();
-		if (YouTube.linkRegex.test(link)) {
-			buf = await YouTube.generateVideoDisplay(link);
-			this.message = this.message.replace(/&ab_channel=(.*)(&|)/ig, '').replace(/https:\/\/www\./ig, '');
+		const SpotifylinkRegex = /open\.spotify\.com\/track\/(.*)/;
+		const SongId = SpotifylinkRegex.exec(link)?.[0];
+		if (YouTube.linkRegex.test(link) || SpotifylinkRegex.test(link)) {
+			if (YouTube.linkRegex.test(link)) {
+				buf = await YouTube.generateVideoDisplay(link);
+				this.message = this.message.replace(/&ab_channel=(.*)(&|)/ig, '').replace(/https:\/\/www\./ig, '');
+			} else {
+				buf = `<spotify src="${SongId}"></spotify>`;
```

Comment:
```ts
		const spotifyRegex = /open\.spotify\.com\/track\/(.*)/;
```
This should probably be at the top level.
```suggestion
		const songId = spotifyRegex.exec(link)?.[0];
		if (YouTube.linkRegex.test(link)) {
			buf = await YouTube.generateVideoDisplay(link);
			this.message = this.message.replace(/&ab_channel=(.*)(&|)/ig, '').replace(/https:\/\/www\./ig, '');
		} else if (songId) {
				buf = `<spotify src="${songId}"></spotify>`;
```

Reply:
```ts
		if (YouTube.linkRegex.test(link) || SpotifylinkRegex.test(link)) {
```
(this isn't necessary when you can just use an else-if, as i put in my above suggestion)